core/src/main/scala/org/broadinstitute/workbench/ccm/CostCalculator.scala

```diff
@@ -0,0 +1,93 @@
+
+  def getPriceOfCall(callMetaDataJson: MetadataResponse, priceList: PriceList): Either[Throwable, Double] = {
+    val ls: List[Either[NonEmptyList[String], Double]] = callMetaDataJson.calls.map { call =>
+      getPriceOfCall(call, priceList, Instant.now(), Instant.now()).leftMap(NonEmptyList.one)
+    }
+
+    ls.parSequence.leftMap(errors => new Exception(errors.toList.mkString(", "))).map(_.sum)
+  }
+
+  private def getPriceOfCall(call: Call, priceList: PriceList, startTime: Instant, endTime: Instant): Either[String, Double] = {
+    for {
+      _ <- if (call.status.asString == "Success") Right(()) else Left(s"Call {name} status was ${call.status.asString}.") // not evaluating workflows that are in flight or Failed or Aborted or whatever
+      diskSize = call.runtimeAttributes.disks.diskSize.asInt + call.runtimeAttributes.bootDiskSizeGb.asInt
+      diskType = call.runtimeAttributes.disks.diskType
+      usageType = getUsageType(call)
+      priceListItem <- Either.catchNonFatal(priceList.prices.get(PriceListKey(call.region, call.machineType, diskType, usageType, false)).get).leftMap(_ => s"couldn't get prices for region ${call.region}, machineType ${call.machineType}, $diskType, $usageType and non-extended.")
+    } yield {
+      // ToDo: calculate subworkflows
+      val wasPreempted = wasCallPreempted(call)
+      // only looking at actual and not requested disk info
+      val callDurationInSeconds = getCallDuration(call, startTime, endTime)
+      // ToDo: add calculating prices for non-custom
+      // adjust the call duration to account for preemptibility
+      // if a VM preempted less than 10 minutes after it is created, user incurs no cost
+      val adjustedCallDurationInSeconds = if (usageType == UsageType.Preemptible && wasPreempted && callDurationInSeconds < (10 * 60)) 0 else callDurationInSeconds
+      val cpuCost = adjustedCallDurationInSeconds * priceListItem.CPUPrice
+      val diskGbHours = call.runtimeAttributes.disks.diskSize.asInt * (adjustedCallDurationInSeconds)
+      val diskCost = diskGbHours * priceListItem.diskCostPerGbPerHour
+      val memCost = adjustedCallDurationInSeconds * priceListItem.RAMPrice
+      cpuCost + diskCost + memCost
+    }
+  }
+
+
+  private def wasCallPreempted(call: Call): Boolean = {
+    // treat preempted and retryableFailure as the same
+    call.executionEvents.exists(event => (event.description.asString.equals("Preempted") || event.description.asString.equals("RetryableFailure")))
+  }
+
+  private def getUsageType(call: Call): UsageType = {
+    if (call.attempt.asInt <= call.runtimeAttributes.preemptibleAttemptsAllowed.asInt)
+      UsageType.Preemptible else UsageType.OnDemand
+    // ToDo: Add false result if the metadata does not contain an "attempt" or preemptible info
+  }
+
+  private def getCallDuration(call: Call, cromwellStartTime: Instant, cromwellEndTime: Instant): Long = {
+    // ToDo: add option to ignore preempted calls and just return 0
```

Comment:
don't preempted calls cost money too?

Reply:
Oh -  this comment was some vestige of a thing we aren't doing anymore (I think earlier we were just going to make the calculation ignore preemptibility for an initial iteration). I'll remove the comment